### ISSUE TITLE
Parameter not being sent to openDBChannel

### ISSUE THREAD

**Issue by tomasfil** [mesqueeb/vuex-easy-firestore#253]
Code calling for open
![image](https://user-images.githubusercontent.com/34779382/66720663-2afac900-ee00-11e9-8607-34f3b9e06852.png)

console output
![image](https://user-images.githubusercontent.com/34779382/66720703-993f8b80-ee00-11e9-9785-31383f234c4c.png)

If I do page refresh, then there is no problem and channel gets opened right.

Org module:
![image](https://user-images.githubusercontent.com/34779382/66720713-c7bd6680-ee00-11e9-8fe2-dc102ecde81d.png)

**Comment by mesqueeb** [comment]
@tomasfil Thank you for this issue!
I confirmed from your screenshots, that indeed it's not working as intended. 🧐

However, I don't have this problem on my end, so I'm not sure what's causing it. (I even have tests for this that pass. 🥺)

Are you on the latest version? → `1.34.3`

I'll have to step through the debugger to find the root I'm afraid. Is there any way you can share your repository?

**Comment by mesqueeb** [comment]
@tomasfil 
The other day it started happening to me as well.
I was able to fix it by calling this BEFORE each `openDBChannel` dispatch:
```js
const myVariable = 'someString'
commit('myVefModule/SET_PATHVARS', {myVariable}, {root: true})
dispatch('myVefModule/openDBChannel', {}, {root: true})
```
I don't know the root of the evil yet, but for now please use this work-around. 😰

**Comment by mesqueeb** [comment]
@tomasfil 
I was able to fix the issue by adding just `commit('myModule/SET_PATHVARS', {myVar})` at first.

However, the other day even with this additional commit before openDBChannel, it still bugged out for me.

I looked into it a little bit more, but it seems that it might have something to do with wether or not Vuex getters are "recalculated" depending on the underlying data.
The bug is present because even though the getter my library uses has its data changed, the getters doesn't always get recalculated... The weird thing is that it's inconsistent, and I'm not sure yet why this is happening.

However, by "console.log"ing the getter it gets recalculated without fail. 😅 
My current work around is as follows:

```js
function openDBChannelsOfCompany (companyId) {
  let path
  const modules = ['myModule1', 'myModule2', 'myModule3'] // etc.
  const promises = productModules.map(moduleName => {
    commit(`${moduleName}/SET_PATHVARS`, {companyId}, {root: true})
    // see: https://github.com/mesqueeb/vuex-easy-firestore/issues/253
    path = rootGetters[`${moduleName}/firestorePathComplete`]
    console.log(path)
    return dispatch(`${moduleName}/openDBChannel`, {companyId}, {root: true})
  })
  Promise.all(promises).then(resolve).catch(reject)
}
```

**Comment by tomasfil** [comment]
Hello, thanks for your effort. I didn´t have too much time to spend on the project, I will have next meeting about it on thursday, so I am happy there is work around and I can continue working (Love your package, helps too much not to use it, will buy u coffee if this project goes live. Its first project using this). Hope so you will find the way to fix this issue. It appeared suddenly out of nowhere. One day it was working and second one did not, not even rebuilding app. I will let you know if anything changes.

**Comment by RichardCr** [comment]
@mesqueeb After much novice spelunking I think I got to the bottom of what is causing it... although I am not quite skilled enough to offer a proper solution. 
In pluginMutations `RESET_VUEX_EASY_FIRESTORE_STATE`
this code kills the observer on the pathVariables, and it never comes back.
```
            var _sync = merge(state._sync, {
                // make null once to be able to overwrite with empty object
                unsubscribe: null,
                pathVariables: null,
                syncStack: { updates: null, propDeletions: null },
                fetched: null,
            }, {
                unsubscribe: {},
                pathVariables: {},
                patching: false,
                syncStack: {
                    inserts: [],
                    updates: {},
                    propDeletions: {},
                    deletions: [],
                    debounceTimer: null,
                },
                fetched: {},
                stopPatchingTimeout: null
            });
```
If you comment out `pathVariables: null,` then the observer is not removed and it works as intended, although this might cause other issues, maybe. Hope this helps with a fix :)

**Comment by tomasfil** [comment]
Tbh, it started somewhere after I started using reset state. This might be actually it!

**Comment by mesqueeb** [comment]
@RichardCr thanks so much, I found the origin of the bug. I am pushing a fix momentarily. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by mesqueeb** [comment]
Good news! I think I was able to fix the bug (with some help ♥) and I just pushed the latest version with the fix: `v1.34.4`

Can you install the latest version; delete any work-arounds you had; and try to see if the problem was fixed?

It should be fixed but maybe some apps still have edge-cases, so let me know how it goes 🙃

<sub>--<br>Vuex Easy Firestore was made with ♥ by Luca Ban.<br>If this library helped you in any way you can support me by buying me <a href="https://www.buymeacoffee.com/mesqueeb">a cup of coffee</a>. ☕️<br>You can also reach out <a href="https://twitter.com/mesqueeb">on twitter</a> if you want a one-on-one coding review/lesson. 🦜</sub>

**Comment by RichardCr** [comment]
Glad to be of some service. Fix works like a charm, thanks for being so quick with the fix! Very appreciated :)
